File: totosuki/ABC/C/449.py
N, L, R = map(int, input().split())
S = input()
cnt = 0
a = ord("a")
data = [[0]*26 for _ in range(N)]

for i in range(N):
    s = ord(S[i])
    if i+L < N:   data[i+L][s-a] += 1
    if i+R+1 < N: data[i+R+1][s-a] -= 1

for i in range(1, N):
    for j in range(26):
        data[i][j] += data[i-1][j]

for i in range(N):
    s = ord(S[i])
    cnt += data[i][s-a]

print(cnt)

File: totosuki/ABC/C/449.rs
use proconio::input;
use proconio::marker::Chars;

fn main() {
    input!{
        N: usize,
        L: usize,
        R: usize,
        S: Chars,
    }

    let mut cnt = 0i64;
    let a = 'a' as usize;
    let mut  data = vec![vec![0; 26]; N];

    

    for i in 0..N {
        let s = S[i] as usize;

        if i+L < N {
            data[i+L][s-a] += 1;
        }

        if i+R+1 < N {
            data[i+R+1][s-a] -= 1;
        }
    }

    for i in 1..N {
        for j in 0..26 {
            data[i][j] += data[i-1][j]
        }
    }

    for i in 0..N {
        let s = S[i] as usize;
        cnt += data[i][s-a];
    }

    println!("{}", cnt);
}
